Update tag.mdx (#4079)
grommet/hpe-design-system: aries-site/src/pages/components/tag.mdx

import {
  TagAnatomy,
  TagMetaData,
  TagSimple,
  CreateTagSimple,
  TagResource,
  TagAttention
} from '../../examples';
import { Example } from '../../layouts';

<Example
  componentName="Tag"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
  docs="https://v2.grommet.io/tag?theme=hpe"
  figma=""
  grommetSource="https://github.com/grommet/grommet/blob/master/src/js/components/Tag/Tag.js"
>
  <TagSimple />
</Example>

## Use cases

### Displaying metadata

> [Static tags](/components/tag#static) can be used to display non critical concise information with

Reviewer: ```suggestion
[Static tags](/components/tag#static) can be used to display non-critical, concise information with
```

the purpose to annotate or label items. Tags can contain value only,
or a name value pair to assist with classification.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <TagMetaData />
</Example>

### Managing metadata

> Tags can be created, assigned and [removed](/components/tag#removable) from resources/items by a user.

Reviewer: ```suggestion
Tags can be created, assigned and [removed](/components/tag#removable) from resources or items by a user.
```

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <CreateTagSimple />
</Example>

### Filtering
[Interactive tags](/components/tag#interactive) can be activated by mouse or keyboard input to enable quick filtering.
Interacting with the tag will display items that share the same assigned value.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <TagResource />
</Example>

### Drawing attention
[Highlight tags](/components/tag#highlight) can be used to draw attention to an element on the page.
They are a decorative variant of the [static tag](/components/tag#static).
Highlights should only contain descriptions that are temporary, such as “New”,

> “Updated”. As such they should not be used a permanent design feature.

Reviewer: ```suggestion
“Updated”. As such they should not be used as a permanent design feature.
```

  pad="small"
  previewWidth="large"
  height={{ min: 'small' }}
>
  <TagAnatomy />
</Example>

| Label | Region        | Purpose                                                      | Required | Notes                                                       |
| :---: | ------------- | ------------------------------------------------------------ | :------: | ----------------------------------------------------------- |
| **1** | **Container** | Controls padding, border & background color.              | Required | Behavior and style will change depending on tag type.                                                        |
| **2** | **Name**      | Descriptive label or category that identifies what kind of data is stored in the associated value. For example “Country” or “Color”.             | Optional | Up to 128 characters max.                     |
| **3** | **Value**     | The specific descriptive data associated with an item.  For example “USA” or “Green”. | Required | Up to 256 characters max. |
| **4** | **Close Icon Button**      | Used to indicate that tag is removable. | Optional | Required for only removable tags.  |

## Types

> ### Tag 

Reviewer: Seems like this should just be "Static", Tag is the thing and static is the type. Then the other types build off of static.

If so, rename this to "Static" and remove the current section called "Static".

- Contains simple, concise information used to annotate, categorize, group or otherwise label items.
- Used to display secondary or supplementary information related to an object or item. 
- Can be used to communicate the state or condition of something (e.g. published, deployed, new, updated, etc.). State is defined [here](/components/notification#taxonomy).
- A tag should not be used to convey system status, that is considered primary information.

### Static 
- All points as above, but static tags are not interactive in any way.

> ### Interactive 

Reviewer: Are we planning to have a visual example here @britt6612? Maybe could use the `plain` treatment like anatomy diagrams

Reviewer 2: yes I was going to add I was waiting on `border` stuff but I can add!

Reviewer 2: added

- Interactive tags enable users to discover, filter, or navigate to identically tagged items.
- An interactive tag can be actioned by mouse or keyboard.

### Removable

> - A removable tag is interactive via the icon button.

Reviewer: ```suggestion
- A removable tag is interactive via the "Close" icon button.
```

- Can be used to dismiss a category/value set from a filtered view.
- Can be used to remove a value that has been assigned to an item or resource.

> ### Highlight

Reviewer: On these bullets, it feels repetitive having "A highlight tag.." can we update to just be

- Used to indicate...
- Draws attention...

Reviewer 2: done

- A highlight tag used to indicate a temporary or transient state associated with an item - eg new, updated, promotional.

> - A highlight tag draws attention but is no louder than a secondary/primary button.

Reviewer: ```suggestion
- A highlight tag draws attention but is no louder than a secondary or primary button.
```

- A highlight tag is not interactive in any way.
- A tags size should be limited to x-small or small.

> - Use “gradientEdge” border for highlight tags. This treatment should be used sparingly.

Reviewer: ```suggestion
- Use a gradient border for highlight tags. This treatment should be used sparingly.
```
The term "gradientEdge" is tool specific. We should make this more generic.

import {
    TagAnatomy,
    TagMetaData,
    TagSimple,
    CreateTagSimple,
    TagResource,
    TagAttention
} from '../../examples';
import { Example } from '../../layouts';

<Example
  componentName="Tag"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
  docs="https://v2.grommet.io/tag?theme=hpe"
  figma=""
  grommetSource="https://github.com/grommet/grommet/blob/master/src/js/components/Tag/Tag.js"
>
  <TagSimple />

> </Example>

Reviewer 2: done

or a name value pair to assist with classification.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <TagMetaData />
</Example>

### Managing metadata
Tags can be created, assigned and removed from resources/items by a user.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <CreateTagSimple />
</Example>

### Filtering
Interactive tags can be activated by mouse or keyboard input to enable quick filtering.
Interacting with the tag will display items that share the same assigned value.

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <TagResource />
</Example>

### Drawing attention
Highlight tags can be used to draw attention to an element on the page.
They are a decorative variant of the static tag.
Highlights should only contain descriptions that are temporary, such as “New”,
“Updated”. As such they should not be used a permanent design feature.

> <Example

Reviewer 2: done

  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/select/SelectExample.js"
>
  <TagAttention />
</Example>

## Anatomy

<Example
  caption="Diagram illustrating the four parts composing tag."
  pad="small"
  previewWidth="large"
  height={{ min: 'small' }}
>
  <TagAnatomy />
</Example>

> | Label | Region        | Purpose                                                      | Required | Notes                                                       |

Reviewer 2: done

| :---: | ------------- | ------------------------------------------------------------ | :------: | ----------------------------------------------------------- |
| **1** | **Container** | Controls padding, border & background color.              | Required | Uses neutral background color for read-only tag and gradient border for marketing tag to differentiate from a button.                                                        |
| **2** | **Name**      | Descriptive label or category that identifies what kind of data is stored in the associated value. For example “Country” or “Color”.             | Optional | --                     |
| **3** | **Value**     | The specific descriptive data associated with an item.  For example “USA” or “Green”. | Required | Required when name item is not available or not needed. |
| **4** | **Close Icon Button**      | Used to indicate that tag is removable. | Optional | Required for only removable tags.  |

## Types

### Tag 
- Contains simple, concise information used to annotate, categorize, group or otherwise label items.
- Used to display secondary or supplementary information related to an object or item. 

> - Can be used to communicate the state or condition of something (e.g. published, deployed, new, updated, etc.). State is defined here.

Reviewer 2: done

- A tag should not be used to convey system status, that is considered primary information.

### Static 

> - All points as above plus...

Reviewer 2: done

- A static tag is not interactive in any way.

### Interactive 
- Interactive tags enable users to discover, filter, or navigate to identically tagged items.
- An interactive tag can be actioned by mouse or keyboard.

### Removable

- A removable tag is interactive via the icon button.
- Can be used to dismiss a category/value set from a filtered view.
- Can be used to remove a value that has been assigned to an item or resource.

### Highlight

> - A highlight tag used to indicate a temporary or transient state associated with an item - eg new, updated, promotional

Reviewer 2: done

> - A highlight tag used to indicate a temporary or transient state associated with an item - eg new, updated, promotional.

Author: e.g. new,
